src/sinks/util/sink.rs

```diff
@@ -129,6 +133,13 @@ impl<T: Sink> Sink for StreamSink<T> {
     }
 }
 
+// === StreamSink2 ===
+
+#[async_trait]
+pub trait StreamSink2 {
```

Comment:
It'd be nice to get rid of the old `StreamSink` and not introduce a `2` here, but I'm not sure how much more work that'd be. Easy to do as a follow up.

Reply:
I'd like not to introduce `StreamSink2` too, but I did this just for the demonstration of the new `StreamSink`. I'll finish update in this PR.

Reply:
In case we can't cut the old `StreamSink` just yet, I propose renaming it to `StreamSinkOld` in this PR.
I figured starting the phasing out with renaming helps a lot with the merges/rebases in the parallel PRs - as the step for that massive rename is pushed to the colder code, rather than the hotter one.

Reply:
Renamed to `StreamSinkOld`. Will rid in other PR. These changes already reviewed/approved, with new changes process will need to be repeated and I also not sure that this will be done before I off for a week.